### DISCUSSION TITLE
Support loading cjs and possibly ts configuration files

### DISCUSSION THREAD

**Discussion by dkrutsko** [directus/directus#17575] [Feature Requests]
### Checklist

- [X] I'm using the [latest version of Directus](https://github.com/directus/directus/releases)
- [X] There's [no other issue](https://github.com/directus/directus/issues?q=is%3Aissue) that already describes the problem.
- [X] I've completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).

### Describe the Bug

Maybe this is more of a feature request however when using `"type": "module"` it becomes more of a bug. Essentially we are limited to only using `yaml`, `json` and `.env` for configuring Directus. However, when trying to set `CONFIG_PATH` to a file ending in the `.cjs` extension, it doesn't read it. I tracked down the problem to [this line](https://github.com/directus/directus/blob/main/api/src/env.ts#L337). Could you please add support to be able to load `cjs` files so that our directus project can use the new module system without eslint complaining about using `require` and `module.exports`.

I would also like to see if Typescript files could be supported for configuration though that would be difficult since it would have to be precompiled. Though I imagine then it would get loaded as an `mjs` file. Alternatively, if you don't want to support `cjs` then how about regular `js` where we are able to use the new `import` syntax.

### To Reproduce

Try setting `export CONFIG_PATH="./config.dev.cjs"` and the configuration won't get loaded.

### Hosting Strategy

Self-Hosted (Custom)

**Comment by br41nslug** [comment]
CommonJS is suported out of the box using the `*.js` extension. Moving this to a feature request for possible extra options 😄

**Reply by dkrutsko** [reply]
Yes it's supported however if I have a project which sets `"type": "module"` in the `package.json`, then node complains about reading that `*.js` file using the `require` keyword. It only works when the `*.cjs` file extension is used. I know it's weird but that's more of a node issue than anything else. Just wanted to point this out as this is what I've been seeing the community do.

Either that, or see if you can move away from using the `require` keyword in the Directus CLI.

**Reply by flevi29** [reply]
If you add another `package.json` in the extension entry point directory or any of its parent directories writing `{"type":"commonjs"}` into it, then it should work. EDIT: However you then need to make sure it's in cjs syntax which is a problem if you're writing typescript. I think Directus API is slowly but surely moving towards being fully ESM though, so maybe one day this will be of minimal hassle.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
